cpp/src/arrow/compute/exec.h

```diff
@@ -36,6 +36,7 @@
 namespace arrow {
 
 class Array;
+class BooleanArray;
```

Comment:
Shouldn't this be in `type_fwd.h`? You could remove all those explicit forward declarations...

Reply:
Note: IWYU unfortunately doesn't recognize forward declarations (https://github.com/include-what-you-use/include-what-you-use/issues/530) but we can add mappings to build-support/iwyu/mappings/arrow-misc.imp to attribute symbols to type_fwd.h so it won't complain about them. I'll fill up these mappings incrementally to help

Reply:
I made some improvements to our IWYU scripts to not complain so much about forward declarations, it should help a lot